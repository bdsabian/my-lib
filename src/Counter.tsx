import * as React from "react";
import { useState } from "react";

interface CounterProps {
  initialCount?: number;
}

const Counter = ({ initialCount = 0 }: CounterProps) => {
  const [count, setCount] = useState(initialCount);

  return (
    <div>
      <p>Count: {count}</p>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </div>
  );
};

export default Counter;
