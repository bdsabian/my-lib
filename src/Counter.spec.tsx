import * as React from "react";
import * as renderer from "react-test-renderer";

import Counter from "./Counter";

test("Counter shows the count", () => {
  const component = renderer.create(<Counter initialCount={2} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
