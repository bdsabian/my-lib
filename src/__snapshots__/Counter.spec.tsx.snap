// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter shows the count 1`] = `
<div>
  <p>
    Count: 
    2
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
